internal-enrichment/virustotal-downloader/src/virustotal-downloader.py

```diff
@@ -0,0 +1,124 @@
+# coding: utf-8
+
+import os
+import yaml
+import time
+import io
+import magic
+import vt
+
+from stix2 import Bundle, Relationship
+
+from pycti import OpenCTIConnectorHelper, OpenCTIStix2Utils, get_config_variable
+
+
+class VirustotalDownloaderConnector:
+    def __init__(self):
+        # Instantiate the connector helper from config
+        config_file_path = os.path.dirname(os.path.abspath(__file__)) + "/config.yml"
+        config = (
+            yaml.load(open(config_file_path), Loader=yaml.FullLoader)
+            if os.path.isfile(config_file_path)
+            else {}
+        )
+        self.helper = OpenCTIConnectorHelper(config)
+
+        self.identity = self.helper.api.identity.create(
+            type="Organization", name="Virustotal", description="Virustotal"
+        )["standard_id"]
+
+        # Get other config values
+        api_key = get_config_variable(
+            "VIRUSTOTAL_DOWNLOADER_API_KEY",
+            ["virustotal_downloader", "api_key"],
+            config,
+        )
+        self.api_client = vt.Client(api_key)
+
+    def _process_hash(self, observable):
+
+        hash_value = observable["observable_value"]
+        artifact_id = None
+        bundle_objects = []
+
+        try:
+
+            # Attempt to source the file contents from the hash
+            bytes_obj = io.BytesIO()
+            response = self.api_client.download_file(hash_value, bytes_obj)
+            bytes_obj.seek(0)
+            file_contents = bytes_obj.read()
+
+            mime_type = magic.from_buffer(file_contents, mime=True)
+
+            # Upload the file as an Artifact
+            kwargs = {
+                "file_name": hash_value,
+                "data": file_contents,
+                "mime_type": mime_type,
+                "x_opencti_description": f"Downloaded from Virustotal via hash {hash_value}",
+            }
+            response = self.helper.api.stix_cyber_observable.upload_artifact(**kwargs)
+
+            self.helper.log_info(response)
+
+            artifact_id = response["standard_id"]
+
+        except Exception as e:
+            raise Exception(
+                f"Failed to download/upload Artifact with hash {hash_value}, exception: {e}"
+            )
+
+        # Create a relationship between the StixFile and the new Artifact
+        relationship = Relationship(
+            id=OpenCTIStix2Utils.generate_random_stix_id("relationship"),
+            relationship_type="related-to",
```

Comment:
Should we in the future (once this PR is merged in the platform https://github.com/OpenCTI-Platform/opencti/pull/1803/files#diff-8f804f0d600cb54e6d42acc99961742916a240a86e8281cac53e41ccb38e16c5R893 ) to change the relationship from `related-to` to Artifact - `content` -> StixFile. As specified here:

content_ref (optional) | identifier | Specifies the content of the file, represented as an Artifact object. The object referenced in this property MUST be of type artifact.
-- | -- | --

https://docs.oasis-open.org/cti/stix/v2.1/os/stix-v2.1-os.html#_vq03pryd7u32

Or should we leave it as it is as a simple `related-to` relationship?